src/main/kotlin/no/nav/familie/ef/iverksett/iverksetting/IverksettService.kt

```diff
@@ -1,28 +1,27 @@
-package no.nav.familie.ef.iverksett.iverksett.start
+package no.nav.familie.ef.iverksett.iverksetting
 
-import no.nav.familie.ef.iverksett.iverksett.domene.Brev
-import no.nav.familie.ef.iverksett.iverksett.domene.Iverksett
-import no.nav.familie.ef.iverksett.iverksett.lagre.LagreIverksettService
-import no.nav.familie.ef.iverksett.iverksett.tilstand.lagre.LagreTilstandService
+import no.nav.familie.ef.iverksett.iverksetting.domene.Brev
+import no.nav.familie.ef.iverksett.iverksetting.domene.Iverksett
+import no.nav.familie.ef.iverksett.iverksettingstatus.status.tilstand.TilstandDbUtil
 import no.nav.familie.ef.iverksett.økonomi.IverksettMotOppdragTask
 import no.nav.familie.prosessering.domene.Task
 import no.nav.familie.prosessering.domene.TaskRepository
 import org.springframework.stereotype.Service
 import java.util.Properties
 
 @Service
-class IverksettService(val taskRepository: TaskRepository, val lagreIverksettService: LagreIverksettService,
-                       val lagreTilstandService: LagreTilstandService) {
+class IverksettService(val taskRepository: TaskRepository, val iverksettDbUtil: IverksettDbUtil,
```

Comment:
Ved å bruke "dbUtil" her, så tar man egentlig inn referanser til infrastrukturkode inn i domenelaget ? Burde dbUtil være en service, eller at den har et interface mellom seg og jdbc implementasjonen ? (separation of concerns)

Reply:
Men et interface som gjør det samme vil jo ikke være noen forskjell mtp separations of concern? Hvis det er begrepet "db" du tenker på som infrastruktur-detalj så er jeg i og for seg enig i det. Da vil isåfall `IverksettRepository` være et bedre begrep - det refererer isåfall bare til en form for lagringsenhet, ikke hvordan det skal lagres.

Reply:
Tenkte på at den "dbUtil" burde evt være en service i alle fall, som ikke bruker jdbc implementasjonen direkte ?

Reply:
Jeg tenker mer at dbUtil er en repository enn service. Den gjør jo kun oppslag mot datalageret og ingenting "service"-basert. Så er `*Jdbc` detaljert implementasjon til hvordan dataene skal lagres. 

Alternativt fjerner vi bare hele `dbUtil` og går direkte mot `*Jdbc`

`IverksettService` sin `startIverksetting` er en service-metode etter min mening. "Lagre, hent, oppdater" mot databasen er ikke nødvendigvis en service, selv om det ofte blir lagt inn i det.

Reply:
Ja, om "dbUtil" enten er et "repository" eller en "service", så var tanken at man burde kanskje som en hovedregel uansett separere dette med et interface for å ikke få referanser til infrastruktur i domenelaget ?

Reply:
But why? Så lenge det kun finnes _en_ implementasjon så er det den vi forholder oss til. Det er uhyre sjelden vi legger inn flere implementasjoner av lagringsenheter - og isåfall er interfacet en ekstra komponent du må trykke deg igjennom?

Reply:
Det er jo for å tydeliggjøre eierskap, og for å holde domenet adskilt fra infrastrukturen, mere enn å tenke at man gjør det for flere "adaptere", kanskje ?

Reply:
Jeg mener at interface ikke adskiller domene og infrastruktur - enten så har du en avhengighet som utfører et stykke arbeid, eller så har du ikke. For å skille de så lager du en egen tjeneste som tar inn data X, utfører endringer og leverer data Y _uten_ å hente/lagre underveis. Da får vi et skille mellom domene og infrastruktur. Så må det en service (med eksplisitt knytning til infrastruktur) til for å hente og persistere data i forkant og etterkant. 

UtbetalingsoppdragGenerator er et eksempel på dette - her har vi skilt domene og infrastruktur. Tasken den kjører i vet meget godt at den må hente data og persistere data etter å ha kjørt denne - det blir ingen forskjell om det er en implementasjon (service, repository+++) eller et interface.

Reply:
Hvis vi klarer å holde alle referanser til infrastrukturspesifikke ting unna domenelaget (jdbc, kafka, dao, controllere, etc), ved at det interfaces til dette, så blir jo dette adskilt, uansett når man henter data fra eller til en av disse ? Det kan jo virke litt pirkete i dette tilfellet kanskje 😄

Reply:
Da er vi nok bare enig om å være uenig på det punktet. 
For meg er 
Infrastrukturen: database, endepunkt til eksterne tjenester, kafkastrømmer, mq-køer osv. Dette innkapsles i en klasse og tilbyr funksjoner som andre klasser kan bruke
Domenelaget: Her behandler vi data på et eller annet nivå
Service: Kobler sammen klassene fra infrastruktur og domenelaget

Uansett om en service tar inn interface eller implementasjon så er det koblet sammen - du kan i teorien bytte ut implementasjonen direkte hvis du har et interface, men det kan du like enkelt gjøre om du bruker implementasjonen.

Reply:
Ja, sant, tilsynelatende ikke så stor forskjell, men implementasjonen må implementere interfacet som "eies" av domenelaget